extensions/soap/src/main/java/jolie/net/SoapProtocol.java

```diff
@@ -697,6 +700,7 @@ public void send_internal( OutputStream ostream, CommMessage message, InputStrea
 
 		final StringBuilder httpMessage = new StringBuilder();
 		ByteArray content = null;
+		String bodyMessage = "???"; // body message for ProtocolMessageEvent
```

Comment:
Why "???" and not just empty ""?

Reply:
The ratio was to signal an anomaly in retriving, in the execution of the code, the body text.
The bodyMessage is setted inside a try ... catch and peraphs an error situation managed by the catch could be throw before the bodyMessage retrieval.

I suggest this change

String bodyMessage = "--- unable to retrieve the body data ---";

Reply:
removed bodyMessage.